### PR TITLE
All features added

### PR DESCRIPTION
**Pull request by Simpleshaikh1**
Create a new Rails app called 'hello-rails-react'.
Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
Verify that the jsbundling-rails gem is added to your Gemfile.
Initialize your project with Git.
Install needed node dependencies as described in the previous lesson's tutorial.
Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
Make sure that your project has a Postgres database set up.
Create a table for storing your messages and create 5 different greetings.
Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
Create a static view that will be the root of your app.
Create the App component with react-router and render it in your static view.
Create the Greeting component that will display a greeting. Set it up as a route in your App component.
Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
Display the random greeting in your Greeting component.

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Simpleshaikh1 ,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that doesn't run. Make sure you fix all the issues and your app can run.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

File: app/controllers/api/greetings_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Api::GreetingsController < ApplicationController
-  def random
-    render json: { greeting: Message.order("RANDOM()").first&.content }
+  def random_greeting
+    @greeting = Message.order('RANDOM()').first
+    render json: { greeting: @greeting.greeting }
```
Comment (Olamarx) [inline app/controllers/api/greetings_controller.rb]: - [ ] I see you added `.greeting` to your "greeting, but there is no greeting object in your seed data, but content, I would encourage you to change it to content or return the whole greeting like that, at the moment, the error below is what I get until I removed the `.greeting`.
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @Simpleshaikh1, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Commit messages meaningful ✔️
- API works ✔️
- Gitflow used ✔️

## Required Changes ♻️
_Check the comments under the review._
- [ ] I am making this comment here because the package.json file is merged. The comment is that react-dom and babel-loader should be part of the dependencies, it means you have not yet added react library and it can't definitely show what is not added, kindly add it too. To prove that your project does not have react at all, the image I circled below shows that react is not featured in the app and that is why it is not on
image
And for GitHub that uses react, you can see in the image below that it is on 
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @Simpleshaikh1, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- "Hello world" project with `react-rails` gem ✔️
- Implement the `API` in the backend correctly ✔️
- Implement the front end correctly ✔️ 
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**